src/main/java/io/jenkins/plugins/opentelemetry/job/log/OtelLogStorage.java

```diff
@@ -48,22 +50,35 @@ public OtelLogStorage(@Nonnull BuildInfo buildInfo, @Nonnull Tracer tracer, @Nul
 
     @Nonnull
     @Override
-    public BuildListener overallListener() {
+    public BuildListener overallListener() throws IOException, InterruptedException {
         OpenTelemetryConfiguration otelConfiguration = JenkinsOpenTelemetryPluginConfiguration.get().toOpenTelemetryConfiguration();
         Map<String, String> otelConfigurationProperties = otelConfiguration.toOpenTelemetryProperties();
         Map<String, String> otelResourceAttributes = new HashMap<>();
         otelConfiguration.toOpenTelemetryResource().getAttributes().asMap().forEach((k, v) -> otelResourceAttributes.put(k.getKey(), v.toString()));
-        return new OtelLogSenderBuildListener.OtelLogSenderBuildListenerOnController(buildInfo, otelConfigurationProperties, otelResourceAttributes);
+
+        OtelLogSenderBuildListener.OtelLogSenderBuildListenerOnController otelLogSenderBuildListenerOnController = new OtelLogSenderBuildListener.OtelLogSenderBuildListenerOnController(buildInfo, otelConfigurationProperties, otelResourceAttributes);
+
+        if (OpenTelemetrySdkProvider.get().isOtelLogsMirrorToDisk()) {
+            File logFile = new File(buildFolderPath, "log");
+            return new MergedBuildListener(otelLogSenderBuildListenerOnController, FileLogStorage.forFile(logFile).overallListener());
+        }
+        return otelLogSenderBuildListenerOnController;
     }
 
     @Nonnull
     @Override
-    public TaskListener nodeListener(@Nonnull FlowNode flowNode) {
+    public TaskListener nodeListener(@Nonnull FlowNode flowNode) throws IOException, InterruptedException {
```

Comment:
I am concerned about these new Exceptions. They are throwing up without any explanation of what happen

Reply:
@kuisathaverat I'm not sure to catch your point. My understanding is that the `IOException` would be thrown by `org.jenkinsci.plugins.workflow.log.FileLogStorage#nodeListener` which sounds reasonable to me. We would have the same exceptions as when using standard log file backed logs.
On the `InterruptedException`, you are right it's not needed.

Did I miss something?

Reply:
my comment was old, the exception is re-throw with a comment now that gives the user some info about what could happen.